src/room.cpp

```diff
@@ -37,8 +37,6 @@ void Room::Destroy() {
         }
         server = nullptr;
     }
-
-    room_thread->join();
```

Comment:
This order of things is actually ok, the thread will keep receiving packets until there's nothing more to receive, then exit the inner loop and then exit the outer loop (because state = Closed), after that we shut down the instance (which should tell the clients that we're shutting down, if i'm reading the docs correctly)